controllers/storageoscluster/controller.go

```diff
@@ -298,6 +302,42 @@ func getCSICondition(ctx context.Context, cl client.Client, namespace string, lo
 	return csiCondition
 }
 
+func getMetricsExporterCondition(ctx context.Context, cl client.Client, namespace string, log logr.Logger) metav1.Condition {
+	metricsCondition := metav1.Condition{
+		Type:    metricsExporterReadyType,
+		Status:  metav1.ConditionFalse,
+		Reason:  notReadyReason,
+		Message: "Metrics-Exporter Not Ready",
+	}
+	nodeDS := &appsv1.DaemonSet{}
+	nodeKey := client.ObjectKey{Name: "storageos-node", Namespace: namespace}
+	if err := cl.Get(ctx, nodeKey, nodeDS); err != nil {
+		if !apierrors.IsNotFound(err) {
+			log.Error(err, "failed to get node status")
```

Comment:
Should we return an error at this point?

Reply:
We only return the condition from this method as per design